### PR TITLE
Add MQTT lawn mower platform

### PR DESCRIPTION
**Pull request by jbouwh**
## Proposed change

Add MQTT lawn mower platform

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [x] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: https://github.com/home-assistant/core/pull/98831
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes #

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

### CODE REVIEW

File: source/_integrations/lawn_mower.mqtt.markdown [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+# Example configuration.yaml entry
+mqtt:
+  - lawn_mower:
+      command_topic: topic
+      name: "Test Lawn Mower"
+```
+
+{% configuration %}
+activity_state_topic:
+  description: The MQTT topic subscribed to receive update of the activity. Valid activities are `mowing`, `paused`, `docked` and `error`. Use `value_template` to extract the activity state from a custom payload. When payload `none` is received the activity state will be reset to `unknown`.
```
Comment (c0ffeeca7) [inline source/_integrations/lawn_mower.mqtt.markdown]:
```suggestion
  description: The MQTT topic subscribed to receive an update of the activity. Valid activities are `mowing`, `paused`, `docked`, and `error`. Use `value_template` to extract the activity state from a custom payload. When payload `none` is received, the activity state will be reset to `unknown`.  
```

File: source/_integrations/lawn_mower.mqtt.markdown [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+      description: The payload that represents the unavailable state.
+      required: false
+      type: string
+      default: offline
+    topic:
+      description: An MQTT topic subscribed to receive availability (online/offline) updates.
+      required: true
+      type: string
+    value_template:
+      description: "Defines a [template](/docs/configuration/templating/#using-templates-with-the-mqtt-integration) to extract device's availability from the `topic`. To determine the devices's availability result of this template will be compared to `payload_available` and `payload_not_available`."
```
Comment (c0ffeeca7) [inline source/_integrations/lawn_mower.mqtt.markdown]:
```suggestion
      description: "Defines a [template](/docs/configuration/templating/#using-templates-with-the-mqtt-integration) to extract the device's availability from the `topic`. To determine the device's availability, the result of this template will be compared to `payload_available` and `payload_not_available`."
```

File: source/_integrations/lawn_mower.mqtt.markdown [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+  description: The MQTT topic subscribed to receive availability (online/offline) updates. Must not be used together with `availability`.
+  required: false
+  type: string
+availability_mode:
+   description: When `availability` is configured, this controls the conditions needed to set the entity to `available`. Valid entries are `all`, `any`, and `latest`. If set to `all`, `payload_available` must be received on all configured availability topics before the entity is marked as online. If set to `any`, `payload_available` must be received on at least one configured availability topic before the entity is marked as online. If set to `latest`, the last `payload_available` or `payload_not_available` received on any configured availability topic controls the availability.
+   required: false
+   type: string
+   default: latest
+availability_template:
+  description: "Defines a [template](/docs/configuration/templating/#using-templates-with-the-mqtt-integration) to extract device's availability from the `availability_topic`. To determine the devices's availability result of this template will be compared to `payload_available` and `payload_not_available`."
```
Comment (c0ffeeca7) [inline source/_integrations/lawn_mower.mqtt.markdown]:
```suggestion
  description: "Defines a [template](/docs/configuration/templating/#using-templates-with-the-mqtt-integration) to extract device's availability from the `availability_topic`. To determine the devices's availability, the result of this template will be compared to `payload_available` and `payload_not_available`."
```

File: source/_integrations/lawn_mower.mqtt.markdown [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+  required: false
+  type: template
+device:
+  description: "Information about the device this Lawn Mower is a part of to tie it into the [device registry](https://developers.home-assistant.io/docs/en/device_registry_index.html). Only works when [`unique_id`](#unique_id) is set. At least one of identifiers or connections must be present to identify the device."
```
Comment (c0ffeeca7) [inline source/_integrations/lawn_mower.mqtt.markdown]:
```suggestion
  description: "Information about the device this lawn mower is a part of to tie it into the [device registry](https://developers.home-assistant.io/docs/en/device_registry_index.html). Only works when the [`unique_id`](#unique_id) is set. At least one of the identifiers or connections must be present to identify the device."
```

File: source/_integrations/lawn_mower.mqtt.markdown [hunk 5]
```diff
@@ -0,0 +1,222 @@
...
+  required: false
+  type: map
+  keys:
+    configuration_url:
+      description: 'A link to the webpage that can manage the configuration of this device. Can be either an `http://`, `https://` or an internal `homeassistant://` URL.'
```
Comment (c0ffeeca7) [inline source/_integrations/lawn_mower.mqtt.markdown]:
```suggestion
      description: 'A link to the webpage that can manage the configuration of this device. Can be either an `http://`, `https://`, or an internal `homeassistant://` URL.'
```

File: source/_integrations/lawn_mower.mqtt.markdown [hunk 6]
```diff
@@ -0,0 +1,222 @@
...
+      required: false
+      type: string
+    connections:
+      description: 'A list of connections of the device to the outside world as a list of tuples `[connection_type, connection_identifier]`. For example the MAC address of a network interface: `"connections": ["mac", "02:5b:26:a8:dc:12"]`.'
```
Comment (c0ffeeca7) [inline source/_integrations/lawn_mower.mqtt.markdown]:
```suggestion
      description: 'A list of connections of the device to the outside world as a list of tuples `[connection_type, connection_identifier]`. For example, the MAC address of a network interface: `"connections": ["mac", "02:5b:26:a8:dc:12"]`.'
```

File: source/_integrations/lawn_mower.mqtt.markdown [hunk 7]
```diff
@@ -0,0 +1,222 @@
...
+      required: false
+      type: list
+    hw_version:
+      description: The hardware version of the device.
+      required: false
+      type: string
+    identifiers:
+      description: 'A list of IDs that uniquely identify the device. For example a serial number.'
```
Comment (c0ffeeca7) [inline source/_integrations/lawn_mower.mqtt.markdown]:
```suggestion
      description: 'A list of IDs that uniquely identify the device. For example, a serial number.'
```

File: source/_integrations/lawn_mower.mqtt.markdown [hunk 8]
```diff
@@ -0,0 +1,222 @@
...
+    suggested_area:
+      description: 'Suggest an area if the device isn’t in one yet.'
+      required: false
+      type: string
+    sw_version:
+      description: The firmware version of the device.
+      required: false
+      type: string
+    via_device:
+      description: 'Identifier of a device that routes messages between this device and Home Assistant. Examples of such devices are hubs, or parent devices of a sub-device. This is used to show device topology in Home Assistant.'
```
Comment (c0ffeeca7) [inline source/_integrations/lawn_mower.mqtt.markdown]:
```suggestion
      description: 'Identifier of a device that routes messages between this device and Home Assistant. Examples of such devices are hubs or parent devices of a sub-device. This is used to show the device topology in Home Assistant.'
```

File: source/_integrations/lawn_mower.mqtt.markdown [hunk 9]
```diff
@@ -0,0 +1,222 @@
...
+  description: The MQTT topic that publishes commands when the service `lawn_mower.dock` service call is executed. The value `dock` is published when the service is called. Use a `dock_command_template` to publish a custom format.
+  required: false
+  type: string
+enabled_by_default:
+  description: Flag which defines if the entity should be enabled when first added.
+  required: false
+  type: boolean
+  default: true
+encoding:
+  description: The encoding of the payloads received and published messages. Set to `""` to disable decoding of incoming payload.
```
Comment (c0ffeeca7) [inline source/_integrations/lawn_mower.mqtt.markdown]:
```suggestion
  description: The encoding of the payloads received and published messages. Set to `""` to disable decoding of the incoming payload.
```

File: source/_integrations/lawn_mower.mqtt.markdown [hunk 10]
```diff
@@ -0,0 +1,222 @@
...
+json_attributes_template:
+  description: "Defines a [template](/docs/configuration/templating/#using-templates-with-the-mqtt-integration) to extract the JSON dictionary from messages received on the `json_attributes_topic`."
+  required: false
+  type: template
+json_attributes_topic:
+  description: The MQTT topic subscribed to receive a JSON dictionary payload and then set as entity attributes. Implies `force_update` of the current activity state when a message is received on this topic.
+  required: false
+  type: string
+name:
+  description: The name of the Lawn Mower. Can be set to `null` if only the device name is relevant.
```
Comment (c0ffeeca7) [inline source/_integrations/lawn_mower.mqtt.markdown]:
```suggestion
  description: The name of the lawn mower. Can be set to `null` if only the device name is relevant.
```

File: source/_integrations/lawn_mower.mqtt.markdown [hunk 11]
```diff
@@ -0,0 +1,222 @@
...
+  description: The MQTT topic that publishes commands when the service `lawn_mower.start_mowing` service call is executed. The value `start_mowing` is published when the service is called. Use a `start_mowing_command_template` to publish a custom format.
+  required: false
+  type: string
+retain:
+  description: If the published message should have the retain flag on or not.
+  required: false
+  type: boolean
+  default: false
+unique_id:
+  description: An ID that uniquely identifies this Lawn Mower. If two Lawn Mower's have the same unique ID Home Assistant will raise an exception.
```
Comment (c0ffeeca7) [inline source/_integrations/lawn_mower.mqtt.markdown]:
```suggestion
  description: An ID that uniquely identifies this lawn mower. If two lawn mowers have the same unique ID, Home Assistant will raise an exception.
```

File: source/_integrations/lawn_mower.mqtt.markdown [hunk 12]
```diff
@@ -0,0 +1,222 @@
+---
+title: "MQTT Lawn Mower"
+description: "Instructions on how to integrate MQTT Lawn Mower's into Home Assistant."
```
Comment (c0ffeeca7) [inline source/_integrations/lawn_mower.mqtt.markdown]:
```suggestion
description: "Instructions on how to integrate MQTT lawn mowers into Home Assistant."
```

File: source/_integrations/lawn_mower.mqtt.markdown [hunk 13]
```diff
@@ -0,0 +1,222 @@
...
+ha_category:
+  - Lawn Mower
+ha_release: 2023.9
+ha_iot_class: Configurable
+ha_domain: mqtt
+---
+
+The `mqtt` `lawn_mower` platform allows control a Lawn Mower over MQTT.
```
Comment (c0ffeeca7) [inline source/_integrations/lawn_mower.mqtt.markdown]:
```suggestion
The `mqtt` `lawn_mower` platform allows controlling a lawn mower over MQTT.
```

File: source/_integrations/lawn_mower.mqtt.markdown [hunk 14]
```diff
@@ -0,0 +1,222 @@
...
+pause_command_template:
+  description: Defines a [template](/docs/configuration/templating/#using-templates-with-the-mqtt-integration) to generate the payload to send to `pause_command_topic`. The `value` parameter in the template will be set to `pause`.
+  required: false
+  type: template
+pause_command_topic:
+  description: The MQTT topic that publishes commands when the service `lawn_mower.pause` service call is executed. The value `pause` is published when the service is called. Use a `pause_command_template` to publish a custom format.
+  required: false
+  type: string
+qos:
+  description: The maximum QoS level of the state topic. Default is 0 and will also be used to publishing messages.
```
Comment (c0ffeeca7) [inline source/_integrations/lawn_mower.mqtt.markdown]: I'm not sure I understand this. Is this saying that 0 is also used to publish messages?

Comment (jbouwh) [reply source/_integrations/lawn_mower.mqtt.markdown]: I'll have a look. I actually copy pasted that part.

Comment (jbouwh) [reply source/_integrations/lawn_mower.mqtt.markdown]: Found some more occurrences that need refrasing, will open a new PR targetting `current` for those once we agree on this one.

File: source/_integrations/lawn_mower.mqtt.markdown [hunk 15]
```diff
@@ -0,0 +1,222 @@
...
+pause_command_template:
+  description: Defines a [template](/docs/configuration/templating/#using-templates-with-the-mqtt-integration) to generate the payload to send to `pause_command_topic`. The `value` parameter in the template will be set to `pause`.
+  required: false
+  type: template
+pause_command_topic:
+  description: The MQTT topic that publishes commands when the service `lawn_mower.pause` service call is executed. The value `pause` is published when the service is called. Use a `pause_command_template` to publish a custom format.
+  required: false
+  type: string
+qos:
+  description: The maximum QoS level of the subscriptions on state topics. This also sets the QoS when publishing messages.
```
Comment (jbouwh) [inline source/_integrations/lawn_mower.mqtt.markdown]:
```suggestion
  description: The maximum QoS level to be used when receiving and publishing messages.
```

**Review by c0ffeeca7 — APPROVED** [review]
LGTM, @jbouwh

**Pull request merged**